### PR TITLE
add native support for Gadfly plots

### PR DESCRIPTION
**Pull request by nomadbl**
Add native Gadfly plot support.
Includes a new usage example and docstrings.

### CODE REVIEW

**Review by oxinabox — APPROVED** [review]
We really should move all these over to Julia 1.9's extension modules / weak deps
Those can work in 1.8 too, so we would only have to restrict compat to  1.8+

I will grant you merge rights shortly so feel free to merge this when happy

**Comment by nomadbl** [comment]
Merging this. There is an unfortunate annoying print to stdout when the plots are displayed, but this is a Gadfly/Compose issue (see https://github.com/GiovineItalia/Gadfly.jl/issues/1621 and https://github.com/GiovineItalia/Compose.jl/issues/440)

**Pull request merged**
